File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/models/ChatCompositeErrorCode.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.chat.models;

import com.azure.android.core.util.ExpandableStringEnum;

/**
 * Defines values for ChatCompositeErrorCode.
 */
public final class ChatCompositeErrorCode extends ExpandableStringEnum<ChatCompositeErrorCode> {
    /**
     * Dispatched when the ACS Token supplied is no longer valid (expired).
     */
    public static final ChatCompositeErrorCode JOIN_FAILED = fromString("joinFailed");
    public static final ChatCompositeErrorCode SEND_MESSAGE_FAILED = fromString("sendMessageFailed");
    public static final ChatCompositeErrorCode START_EVENT_NOTIFICATIONS_FAILED =
            fromString("startEventNotificationsFailed");
    public static final ChatCompositeErrorCode FETCH_MESSAGES_FAILED = fromString("fetchMessagesFailed");
    public static final ChatCompositeErrorCode REQUEST_PARTICIPANTS_FETCH_FAILED =
            fromString("requestParticipantsFetchFailed");
    public static final ChatCompositeErrorCode SEND_READ_RECEIPT_FAILED = fromString("sendReadReceiptFailed");
    public static final ChatCompositeErrorCode SEND_TYPING_INDICATOR_FAILED =
            fromString("sendTypingIndicatorFailed");

    private static ChatCompositeErrorCode fromString(final String name) {
        return fromString(name, ChatCompositeErrorCode.class);
    }
}

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/error/ChatCompositeErrorCode.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.chat.error

import com.azure.android.core.util.ExpandableStringEnum

internal class ChatCompositeErrorCode : ExpandableStringEnum<ChatCompositeErrorCode?>() {
    companion object {
        val CHAT_JOIN_FAILED = fromString("chatJoinFailed")
        val CHAT_SEND_MESSAGE_FAILED = fromString("chatSendMessageFailed")
        val CHAT_START_EVENT_NOTIFICATIONS_FAILED = fromString("chatStartEventNotificationsFailed")
        val CHAT_FETCH_MESSAGES_FAILED = fromString("chatFetchMessagesFailed")
        val CHAT_REQUEST_PARTICIPANTS_FETCH_FAILED =
            fromString("chatRequestParticipantsFetchFailed")
        val CHAT_SEND_EDIT_MESSAGE_FAILED = fromString("chatSendEditMessageFailed")
        val CHAT_SEND_READ_RECEIPT_FAILED = fromString("chatSendReadReceiptFailed")
        val CHAT_SEND_TYPING_INDICATOR_FAILED = fromString("chatSendTypingIndicatorFailed")

        private fun fromString(name: String): ChatCompositeErrorCode {
            return fromString(name, ChatCompositeErrorCode::class.java)
        }
    }
}
